subworkflows/kraken_pipeline.nf

```diff
@@ -9,42 +9,49 @@ workflow kraken_pipeline {
-            keys = sources.keySet()
-            throw new Exception("Source $params.source is invalid, must be one of $keys")
-        }
+        // sources = params.database_sets
+        // source_name = params.database_set
+        // source_data = sources.get(source_name, false)
+        // if (!sources.containsKey(source_name) || !source_data) {
+        //     keys = sources.keySet()
+        //     throw new Exception("Source $params.source is invalid, must be one of $keys")
+        // }
 
         // Grab taxonomy files
-        taxonomy = file(sources[source_name]["taxonomy"], type: "file")
-        if (params.taxonomy) {
-            log.info("Checking custom taxonomy mapping exists")
-            taxonomy = file(params.taxonomy, type: "dir", checkIfExists:true)
-        }
+        // taxonomy = file(sources[source_name]["taxonomy"], type: "file")
+        // if (params.taxonomy) {
+        //     log.info("Checking custom taxonomy mapping exists")
+            
+        // }
+
+        taxonomy = file(params.taxonomy_dir, type: "dir", checkIfExists:true)
 
-        if (database) {
-            source_database = database
-        } else {
-            source_database = source_data.get("database", false)
-            if (!source_database) {
-                throw new Exception(
-                    "Error: Source $source_name does not include a database for "
-                    + "use with kraken, please choose another source, "
-                    + "provide a custom database or disable kraken.")
-            }
-        database = file(source_database, type: "file")
+        // if (database) {
+        //     source_database = database
+        // } else {
+        //     source_database = source_data.get("database", false)
+        //     if (!source_database) {
+        //         throw new Exception(
+        //             "Error: Source $source_name does not include a database for "
+        //             + "use with kraken, please choose another source, "
+        //             + "provide a custom database or disable kraken.")
+        //     }
+        database = file(params.db, type: "dir", checkIfExists:true)
 
         // start_server(database, taxonomy)
         run_kraken_and_bracken(unique_id, fastq, database, taxonomy)
+
+        run_kraken_and_bracken.out.json
+            .collect()
+            .set { bracken_jsons }
+
         // stop_server(run_kraken_and_bracken.out.bracken_report.collect())
         qc_checks(unique_id, fastq)
-        all_bracken_jsons = Channel.fromPath("${params.out_dir}/${unique_id}/classifications/*.bracken.json")
-                    .concat(run_kraken_and_bracken.out.json)
-                    .unique {it.getName()}
-                    .collect()
-        generate_report(unique_id, qc_checks.out, all_bracken_jsons )
+        // all_bracken_jsons = Channel.fromPath("${params.out_dir}/${unique_id}/classifications/*.bracken.json")
```

Comment:
I'm almost certain I've broken something here but going outside of the work directory to collect files is something we can't do in a cloud execution env like this